### PR TITLE
feat: hook up sentry boundary

### PR DESCRIPTION
**Pull request by P0lip**
Try it out live here https://github.com/stoplightio/platform-internal/pull/8151

### CODE REVIEW

File: src/ErrorBoundary.tsx [hunk 1]
```diff
@@ -1,89 +1,49 @@
...
-      props: { FallbackComponent: Fallback = this.context.FallbackComponent || FallbackComponent, children },
-      state: { error, componentStack },
-    } = this;
+  React.useEffect(() => {
+    const boundary = boundaryRef.current;
+    if (!boundary || !prevProps || !boundary.state.error || !props.recoverableProps) return;
 
-    if (error !== null) {
-      return <Fallback error={error} componentStack={componentStack} tryRecovering={this.recover} />;
+    for (const recoverableProp of props.recoverableProps) {
```
Comment (P0lip) [inline src/ErrorBoundary.tsx:37]: @marbemac @lottamus whatcha think about dropping `recoverableProps`?
I know it's used here and there, but one can use the `reload` button in the fallback if needed.

This seems to work reasonably though, so can keep it.

File: src/ErrorBoundary.tsx [hunk 2]
```diff
@@ -1,89 +1,49 @@
...
-import { ErrorBoundaryProps, ErrorBoundaryState } from './types';
+import { ErrorBoundaryProps, FallbackProps } from './types';
 
-export class ErrorBoundary<P extends object = {}> extends React.PureComponent<
-  P & ErrorBoundaryProps<P>,
-  ErrorBoundaryState
-> {
-  public static contextType = ErrorBoundaryContext;
-  public context!: React.ContextType<typeof ErrorBoundaryContext>;
+const wrapFallback = (Component: React.ElementType<FallbackProps>): FallbackRender => {
```
Comment (P0lip) [inline src/ErrorBoundary.tsx:8]: this is to mitigate the breaking changes.
In fact there are almost no breaking changes introduced in this whole PR, so the bump should be straightforward.

File: src/ErrorBoundaryProvider.tsx [hunk 3]
```diff
@@ -1,15 +1,12 @@
-import { ICoreReportingAPI } from '@stoplight/reporter';
 import * as React from 'react';
 import { FallbackComponent } from './FallbackComponent';
 import { FallbackProps } from './types';
 
 export type ErrorBoundaryContext = {
-  reporter: ICoreReportingAPI;
```
Comment (P0lip) [inline src/ErrorBoundaryProvider.tsx:7]: this is probably the only notable breaking change

File: src/types.ts [hunk 4]
```diff
@@ -18,7 +18,6 @@ export type ErrorBoundaryProps<P extends object = {}> = {
 
   // these are our own additions
   recoverableProps?: Array<keyof P>;
-  reportErrors?: boolean; // true by default
```
Comment (P0lip) [inline src/types.ts:21]: and this is the only change in props.

File: src/ErrorBoundaryProvider.tsx [hunk 5]
```diff
@@ -1,15 +1,12 @@
...
   FallbackComponent?: React.ElementType<FallbackProps>;
 };
 
 export const ErrorBoundaryContext = React.createContext<ErrorBoundaryContext>({
```
Comment (P0lip) [inline src/ErrorBoundaryProvider.tsx:9]: One more note - I believe we don't quite need it. While having a custom default fallback might be nice to have, I believe the existing fallback we use is good enough, and we don't really need to override it globally.

Comment (P0lip) [reply src/ErrorBoundaryProvider.tsx:9]: I got rid of it. Let's have simpler code.

File: src/ErrorBoundary.tsx (hunk 1 above)
Comment (marbemac) [reply src/ErrorBoundary.tsx:37]: How "in use" is `recoverableProps` in our ecosystem? What are the benefit(s) of it versus just leveraging the out of the box `fallback` + `resetError` and `onReset` functionality that sentry error boundary exposes out of the box?

Comment (marbemac) [reply src/ErrorBoundary.tsx:37]: Ah no actually this is a pretty good feature IMHO - would be nice to keep if it doesn't introduce much complexity etc.

File: src/ErrorBoundaryProvider.tsx (hunk 5 above)
Comment (P0lip) [reply src/ErrorBoundaryProvider.tsx:9]: Ah, we actually have some custom usage of it in here packages/ninja/src/components/Fallback.tsx
https://github.com/stoplightio/platform-internal/blob/v2/packages/ninja/src/components/Fallback.tsx.
Alright, gonna keep it. Can delete later on if needed. It's actually prob nicer to have "contact Stoplight team" in the context of Platform, but less so overall (public usage of Elements, etc)

File: src/ErrorBoundary.tsx (hunk 1 above)
Comment (P0lip) [reply src/ErrorBoundary.tsx:37]: It doesn't - and it seems to be working correctly, verified it in json-schema-viewer, platform-internal & the storybook here in react-error-boundary.

**Review by marbemac — APPROVED** [review]
Make sure to include breaking change or whatever it is in the merge so that it releases as a major. And perhaps update the release notes after to just note the breaking changes clearly.

**Pull request merged**